### PR TITLE
Support lists/arrays on entity editor

### PR DESCRIPTION
**Pull request by yusufkinatas**
## 🌟 What is the purpose of this PR?

This PR adds list/array editing support on Entity editor. There will be follow-up PRs to add multi-type support and UX improvements to this.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203150072758509/1203199644802788/f) _(internal)_
- [Design](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH-App?node-id=6295%3A253013&t=AcMTJsXFjdQWI8tA-4) _(internal)_

## 🔍 What does this change?

- `PropertyRow`
  - add `isArray`
  - rename `dataTypes` to `expectedTypes`
  - fix `expectedTypes`, now is shows types expected types in an array instead of showing `Array` as type
- Add `bpList` icon to `custom-grid-icons`

## 🐾 Next steps

- Improve mobile UX (e.g. test drag-drop on mobile)
- Support multiple types on array-editor
- Lock page scroll when the editor is open

## ❓ How to test this?

1.  Checkout the branch / view the deployment
1.  Apply https://github.com/hashintel/hash/commit/19c2f7fd0b4426e69ac744848fcf63b5b7089b2f this commit on your local to get a button on entity editor, which creates a complex entity type for you to test this 
1.  Confirm that we can edit lists on entity editor

## 📹 Demo

⚠️ Ignore the gray page opens at 0:42, that's the break-reminder app on my computer 😄 

https://user-images.githubusercontent.com/39553853/203017102-84971856-1af2-4a01-9b0c-1440e4119062.mov

### CODE REVIEW

File: packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/entity-editor/properties-section/property-table/cells/value-cell/value-cell-editor/array-editor.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+      ];
+    });
+    onChange(newCell);
+  };
+
+  const removeItem = (index: number) => {
+    const newCell = produce(cell, (draftCell) => {
+      draftCell.data.property.value = items
+        .filter((_, index2) => index !== index2)
+        .map(({ value }) => value);
```
Comment (CiaranMn) [inline packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/entity-editor/properties-section/property-table/cells/value-cell/value-cell-editor/array-editor.tsx:63]: I think this makes it a bit clearer
```suggestion
  const removeItem = (indexToRemove: number) => {
    const newCell = produce(cell, (draftCell) => {
      draftCell.data.property.value = items
        .filter((_, index) => indexToRemove !== index)
        .map(({ value }) => value);
```

File: packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/entity-editor/properties-section/property-table/cells/value-cell/value-cell-editor/array-editor.tsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+  const updateItem = (index: number, value: string) => {
+    setEditingRow("");
+
+    if (!value.trim().length) {
+      return removeItem(index);
+    }
+
+    const newCell = produce(cell, (draftCell) => {
+      draftCell.data.property.value = items.map((item, index2) =>
+        index === index2 ? value : item.value,
```
Comment (CiaranMn) [inline packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/entity-editor/properties-section/property-table/cells/value-cell/value-cell-editor/array-editor.tsx:77]:
```suggestion
  const updateItem = (indexToUpdate: number, value: string) => {
    setEditingRow("");

    if (!value.trim().length) {
      return removeItem(indexToUpdate);
    }

    const newCell = produce(cell, (draftCell) => {
      draftCell.data.property.value = items.map((item, index) =>
        indexToUpdate === index ? value : item.value,
```

File: packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/entity-editor/properties-section/property-table/cells/value-cell/value-cell-editor/array-editor.tsx [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        overflow: "hidden",
+      })}
+    >
+      <Box
+        ref={scrollableContainer}
+        sx={{
+          maxHeight: 300,
+          overflowY: "scroll",
+          overflowX: "hidden",
+          borderBottom: `1px solid`,
```
Comment (CiaranMn) [inline packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/entity-editor/properties-section/property-table/cells/value-cell/value-cell-editor/array-editor.tsx:117]:
```suggestion
          borderBottom: "1px solid",
```

File: packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/entity-editor/properties-section/property-table/cells/value-cell/value-cell-editor/array-editor/inline-text-editor.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { TextField } from "@hashintel/hash-design-system";
+
+export const InlineTextEditor = ({
```
Comment (CiaranMn) [inline packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/entity-editor/properties-section/property-table/cells/value-cell/value-cell-editor/array-editor/inline-text-editor.tsx:3]: What about numbers? It looks like the array editor is only doing strings at the moment

Comment (yusufkinatas) [reply packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/entity-editor/properties-section/property-table/cells/value-cell/value-cell-editor/array-editor/inline-text-editor.tsx:3]: Yes, it's only doing the strings at the moment, gonna support other types in follow-up PRs as I mentioned in the PR description

Comment (CiaranMn) [reply packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/entity-editor/properties-section/property-table/cells/value-cell/value-cell-editor/array-editor/inline-text-editor.tsx:3]: Ah, okay – I thought by "support multiple types" you meant supporting arrays of mixed types (e.g. `(string | number)[]`

**Comment by yusufkinatas** [comment]
@CiaranMn thanks for the suggestions, addressed them!

**Review by CiaranMn — APPROVED** [review]

**Pull request merged**